python_scripts/functions.py

```diff
@@ -408,27 +411,49 @@ def generate_iam_config(
         Optional path to output the iam policy json generated from the iam_config just generated
     """
 
+    if os.path.exists(iam_config_output) and overwrite_config is not True:
+        raise ValueError(
+            f"{iam_config_output} exists: to overwrite set overwrite_config=True"
+        )
+
+    log_path = config["log-base-path"].replace("s3://", "")
+    land_path = config["land-base-path"].replace("s3://", "")
+    pass_path = config["pass-base-path"].replace("s3://", "")
+    if config["fail-base-path"]:
+        fail_path = config["fail-base-path"].replace("s3://", "")
+    else:
+        fail_path = config["fail-base-path"]
+
+    if fail_path:
+        read_write = [
+                os.path.join(land_path, "*"),
+                os.path.join(pass_path, "*"),
+                os.path.join(fail_path, "*"),
+            ]
+    else:
+        read_write = [
+                os.path.join(land_path, "*"),
+                os.path.join(pass_path, "*")
+            ]
+
```

Comment:
Is this equivalent?
```
read_write = [
                os.path.join(land_path, "*"),
                os.path.join(pass_path, "*")
            ]

if "fail-base-path" in config:
    fail_path = config["fail-base-path"].replace("s3://", "")
    read_write.append(os.path.join(fail_path, "*"))
```

Reply:
fail base path has to be in the config but it can be set to be `null`